src/basicfuns.jl

```diff
@@ -149,6 +149,31 @@ end
 """
 $(SIGNATURES)
 
+Return `log(abs(tanh(x)))`, carefully evaluated without intermediate calculation of `tanh(x)`.
+
+The implementation ensures `logabstanh(-x) = logabstanh(x)`.
+"""
+function logabstanh(x::Real)
+    return log1p(-2/((exp(2abs(x))+1)))
```

Comment:
Probably the safer default would be just

```suggestion
    return log(abs(tanh(x)))
```